### DISCUSSION TITLE
Passing an incorrect operator when querying the REST API returns ALL documents

### DISCUSSION THREAD

**Discussion by rbruels** [payloadcms/payload#5008] [Feature Requests & Ideas]
### Describe the Bug

Passing an incorrect operator (i.e. an operator not defined in [types/constants.ts](https://github.com/payloadcms/payload/blob/b30ea8aa6bd31adb9d2e112505b247099e726ce1/packages/payload/src/types/constants.ts#L1)) does not fail the query -- in fact, it returns all documents in that collection (!).

I would expect the query to fail — given the `where` it's clear that I'm trying to query the collection and not just specifying random query params.

### To Reproduce

Assuming `collection` has this data in it:
```
[
   { id: "ABC" },
   { id: "DEF" },
   { id: "GHI" }
]

`https://my.test.cms/api/collection?where[id][equals]=XYZ`
correctly returns zero documents

`https://my.test.cms/api/collection?where[id][equasl]=XYZ`
incorrectly returns _all_ documents

### Payload Version

2.4.0

### Adapters and Plugins

db-mongodb

**Comment by jmikrut** [comment]
Hey @rbruels — this is actually expected - we just disregard query parameters that we do not recognize at this time. But, we can update this logic to throw a `400` or similar status code in the future if you provide an operator that does not exist.

I will convert this to a feature request and we will track it accordingly there. Thank you for bringing this up!
